src/adapter/debugAdapter.ts

```diff
@@ -136,9 +136,25 @@ export class DebugAdapter implements IDisposable {
   private async _getPreferredUILocation(
     params: Dap.GetPreferredUILocationParams,
   ): Promise<Dap.GetPreferredUILocationResult> {
-    const source = this.sourceContainer.source(params.source);
+    let source: Source | undefined = undefined;
+    if (params.originalUrl) {
+      source = this.sourceContainer.getSourceByOriginalUrl(params.originalUrl);
+    }
+    if (!source && params.source) {
+      source = this.sourceContainer.source(params.source);
+    }
+
     if (!source) {
-      return params;
+      if (params.source) {
+        // Return unmodified input source
+        return {
+          column: params.column,
+          line: params.line,
+          source: params.source,
+        };
+      } else {
+        throw new Error('Missing source information');
```

Comment:
You should make a new error in `dap/errors.ts` and throw that so it gets an error code and is typed nicely as an expected error